src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,523 @@
+> events too often.
+>
+> Generally you should always use external power when using Bluetooth
+> or connect something like a servo to your board.
+
+With noble creating a characteristic is quite easy. All you need to do
+is to define how the characteristic communicates (options are read,
+write and notify and any combination thereof) as well as define an uuid.
+Creating a new object and inheriting from the bleno.Characteristic is
+the easiest way.
+
+> NOTE: I am not using ES2016 here as Edison currently uses an older
+> version of Node.JS.
+>
+> With the newly launched [Ostro Project](https://ostroproject.org)
+> which supports the Edison, that is no longer the case. Also, if you
+> use Brillo as part of the Brillo Early Access Program, then it is
+> possible to compile and install a recent version of Node.JS.
+
+The resulting characteristic object looks like the following:
+
+{% highlight javascript %}
+var TemperatureCharacteristic = function() {
+  bleno.Characteristic.call(this, {
+    uuid: 'fc0a',
+    properties: ['read', 'notify'],
+    value: null
+  });
+    
+  this._lastValue = 0;
+  this._total = 0;
+  this._samples = 0;
+  this._onChange = null;
+};
+
+util.inherits(TemperatureCharacteristic, bleno.Characteristic);
+{% endhighlight %}
+
+We are storing the current temperature value in the `this._lastValue`
+variable. In order for a "read" to work we need to add an onReadRequest
+method and encode the value.
+
+{% highlight javascript %}
+TemperatureCharacteristic.prototype.onReadRequest = function(offset, callback) {
+  var data = new Buffer(8);
+  data.writeDoubleLE(this._lastValue, 0);
+  callback(this.RESULT_SUCCESS, data);
+};
+{% endhighlight %}
+
+For "notify" we need to add a method to handle subscription and
+unsubscription. Basically, we simple store a callback. When we
+have a new temperature reason we want to send, we then call that
+callback with the new value (encoded as above).
+
+{% highlight javascript %}
+TemperatureCharacteristic.prototype.onSubscribe = function(maxValueSize, updateValueCallback) {
+  console.log("Subscribed to temperature change.");
+  this._onChange = updateValueCallback;
+  this._lastValue = undefined;
+};
+
+TemperatureCharacteristic.prototype.onUnsubscribe = function() {
+  console.log("Unsubscribed to temperature change.");
+  this._onChange = null;
+};
+{% endhighlight %}
+
+As values can fluctuate a bit, we need to smoothen the values we
+get from the TMP36 sensor. I opted to simple take the average over
+100 reasons and only send updates when the temperature changes by
+at least 1 degree.
+
+{% highlight javascript %}
+TemperatureCharacteristic.prototype.valueChange = function(value) {
+    this._total += value;
+    this._samples++;
+    
+    if (this._samples < NO_SAMPLES)
+        return;
+        
+    var newValue = Math.round(this._total / NO_SAMPLES);
+    
+    this._total = 0;
+    this._samples = 0;
+    
+    if (this._lastValue && Math.abs(this._lastValue - newValue) < 1)
```

Comment:
I'd personally wrap these in  `{}` to make it easier for novices to understand it, especially when a copy and paste screws with tabs etc and their example doesnt work...